### ISSUE TITLE
Improve Route53/DNS provider auditability

### ISSUE THREAD

**Issue by jasonmadigan** [Kuadrant/kcp-glbc#236]
**What**

Say I'm an administrator, running the GLBC. I've got access to Route53 for the account hosting the GLBC's DNS. I'd like for there to be a way for me to map Route53 resources (hosted zones, and records) to GLBC-supported workspaces and workloads.

An example: I'm an SRE, and I can see the GLBC has created a host for my service. I'd like to be able to map that hosted zone to a specific cluster.

**How**
[ ] Add descriptions for hosted zones with extra meta-data that describes what workspace this zone is associated with
[ ] For Records, add a comment with meta-data describing the home/user of the resource

**Comment by jasonmadigan** [comment]
cc'ing @Kuadrant/hcg-engineering

**Comment by david-martin** [comment]
> * Add descriptions for hosted zones with extra meta-data that describes what workspace this zone is associated with

Set a value in the 'Description' field on the Hosted Zone in route53?

> * For Records, add a comment with meta-data describing the home/user of the resource

Is this something in route53 you can do as well, 
or do you mean add info to the DNSRecord CR status block?

**Comment by jasonmadigan** [comment]
> > * Add descriptions for hosted zones with extra meta-data that describes what workspace this zone is associated with
> 
> Set a value in the 'Description' field on the Hosted Zone in route53?

Exactly yep.
> 
> > * For Records, add a comment with meta-data describing the home/user of the resource
> 
> Is this something in route53 you can do as well, or do you mean add info to the DNSRecord CR status block?

Probably need to look into this one - looking at the record docs, there seemed to be several options (identifiers, comments)



> > * Add descriptions for hosted zones with extra meta-data that describes what workspace this zone is associated with
> 
> Set a value in the 'Description' field on the Hosted Zone in route53?
> 
> > * For Records, add a comment with meta-data describing the home/user of the resource
> 
> Is this something in route53 you can do as well, or do you mean add info to the DNSRecord CR status block?

**Comment by jasonmadigan** [comment]
Yeah, like you I couldn't see a way to use normal AWS tags for Route53 resources, but I did see some fields like identifiers/notes/descriptions for some of the resources (hosts and records). We'd need to do a bit of digging to spec out what we could use.

@mikenairn the docs in this case would be the R53 ones
